File: content/problems/1585-check-if-string-is-transformable-with-substring-sort-operations/solutions/digit-position-queues/solution.cpp
class Solution {
    public: bool isTransformable(string s,string t){
        array<queue<int>,10> positions;
        for(int i=0;i<(int)s.size();++i)positions[s[i]-'0'].push(i);
        for(char value:t){
            int digit=value-'0';
            if(positions[digit].empty())return false;
            int position=positions[digit].front();
            for(int smaller=0;smaller<digit;++smaller)if(!positions[smaller].empty()&&positions[smaller].front()<position)return false;
            positions[digit].pop();
        }
        return true;
    }
};

File: content/problems/1585-check-if-string-is-transformable-with-substring-sort-operations/solutions/digit-position-queues/Solution.kt
class Solution {
    fun isTransformable(s:String,t:String):Boolean{
        val positions=Array(10){
            java.util.ArrayDeque<Int>()
        };
        for(i in s.indices)positions[s[i]-'0'].addLast(i);
        for(value in t){
            val digit=value-'0';
            if(positions[digit].isEmpty())return false;
            val position=positions[digit].peekFirst();
            for(smaller in 0 until digit)if(positions[smaller].isNotEmpty()&&positions[smaller].peekFirst()<position)return false;
            positions[digit].removeFirst()
        };
        return true
    }
}
